collectors/mesos/agent/agent.go

```diff
@@ -123,6 +123,19 @@ func (c *Collector) metricsMessages() (out []producers.MetricsMessage) {
 			continue
 		}
 
+		ei, ok := getExecutorInfoByExecutorID(cm.ExecutorID, fi.Executors)
+		if !ok {
+			c.log.Warnf("Did not find ExecutorInfo for executor ID %s, skipping!", cm.ExecutorID)
+			continue
+		}
+
+		ti, ok := getTaskInfoByContainerID(cm.ContainerID, ei.Tasks)
+		if !ok {
+			// This is not a warning because task ID is not guaranteed to be set, eg
+			// a custom executor is a container with no associated task.
+			c.log.Debugf("Did not find TaskInfo for container ID %s, skipping!", cm.ContainerID)
```

Comment:
A container should always have an associated executor and framework, it's just the task association that isn't guaranteed. I think it's correct to skip and warn if no associated executor is found, because otherwise we'd be serving an unidentifiable metric.